runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/RegisterAndProcessBundleOperation.java

```diff
@@ -446,7 +447,11 @@ long getInputElementsConsumed(final Iterable<MonitoringInfo> monitoringInfos) {
         String pcollection =
             mi.getLabelsOrDefault(MonitoringInfoConstants.Labels.PCOLLECTION, null);
         if (pcollection != null && pcollection.equals(grpcReadTransformOutputPCollectionName)) {
-          return mi.getMetric().getCounterData().getInt64Value();
+          try {
+            return VARINT_CODER.decode(mi.getPayload().newInput());
```

Comment:
ditto, this is also too low level, please use a helper here. Let's try to encapsulate the bytes payload format of MonitoringInfos.